File: android/src/main/java/me/timeto/app/ui/notes/NotesView.kt
package me.timeto.app.ui.notes

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.defaultMinSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyListState
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import me.timeto.app.ui.Divider
import me.timeto.app.ui.ZStack
import me.timeto.app.ui.c
import me.timeto.app.ui.home.HomeScreen__itemHeight
import me.timeto.app.ui.navigation.LocalNavigationFs
import me.timeto.app.ui.rememberVm
import me.timeto.shared.db.NoteFolderDb
import me.timeto.shared.vm.notes.NotesVm

@Composable
fun NotesView(
    noteFolderDb: NoteFolderDb,
    hPadding: Dp,
    scrollState: LazyListState,
    withDivider: Boolean,
    modifier: Modifier,
) {

    val (_, state) = rememberVm {
        NotesVm(
            noteFolderDb = noteFolderDb,
        )
    }

    val navigationFs = LocalNavigationFs.current
    val notesUi = state.notesUi.reversed()

    LazyColumn(
        modifier = modifier,
        state = scrollState,
        reverseLayout = true,
    ) {
        notesUi.forEach { noteUi ->
            item {
                ZStack(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable {
                            navigationFs.push {
                                NoteFs(
                                    initNoteDb = noteUi.noteDb,
                                )
                            }
                        }
                        .defaultMinSize(minHeight = HomeScreen__itemHeight),
                ) {
                    if (withDivider && (notesUi.last() != noteUi)) {
                        Divider(
                            modifier = Modifier
                                .padding(start = hPadding),
                        )
                    }

                    Text(
                        text = noteUi.text,
                        modifier = Modifier
                            .padding(horizontal = hPadding, vertical = 6.dp),
                        color = c.white,
                        maxLines = 3,
                        overflow = TextOverflow.Ellipsis,
                    )
                }
            }
        }
    }
}

File: apple/iphone/UI/Notes/NotesView.swift
import SwiftUI
import shared

struct NotesView: View {
    
    let noteFolderDb: NoteFolderDb
    let hPadding: CGFloat
    let withDivider: Bool
    
    var body: some View {
        VmView({
            NotesVm(
                noteFolderDb: noteFolderDb,
            )
        }) { vm, state in
            let state = vm.state.value as! NotesVm.State
            NotesViewLocal(
                state: state,
                notesUi: state.notesUi,
                hPadding: hPadding,
                withDivider: withDivider,
            )
        }
    }
}

private struct NotesViewLocal: View {
    
    let state: NotesVm.State
    let notesUi: [NotesVm.NoteUi]
    let hPadding: CGFloat
    let withDivider: Bool

    ///
    
    @Environment(Navigation.self) private var navigation
    
    var body: some View {
        ScrollView(showsIndicators: false) {
            VStack {
                ForEach(notesUi, id: \.noteDb.id) { noteUi in
                    Button(
                        action: {
                            navigation.push(.note(noteDb: noteUi.noteDb, onDelete: {}))
                        },
                        label: {
                            ZStack {
                                if withDivider && (notesUi.first != noteUi) {
                                    VStack {
                                        Divider()
                                            .padding(.leading, hPadding)
                                        Spacer()
                                    }
                                }
                                
                                Text(noteUi.text)
                                    .foregroundColor(.white)
                                    .lineLimit(3)
                                    .padding(.horizontal, hPadding)
                                    .padding(.vertical, 4)
                                    .textAlign(.leading)
                            }
                            .frame(minHeight: HomeScreen__itemHeight)
                        },
                    )
                }
            }
        }
        .defaultScrollAnchor(.bottom)
    }
}
